showcase/echo_test.go

```diff
@@ -94,6 +97,63 @@ func TestEcho_error(t *testing.T) {
 
 }
 
+// Test dynamic routing header generation.
+func TestEchoHeader(t *testing.T) {
+	defer check(t)
+	content := "hello world!"
+	reqs := []*showcasepb.EchoRequest{&showcasepb.EchoRequest{
```

Comment:
Take a look at this piece about [Table Driven Tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) and look elsewhere in our codebase. Let's refactor this list of requests into a list of test cases with the expected headers included so we can say "for this request, I expect this header".

Reply:
Thanks for sharing! Makes sense. I've refactored the test a bit (it also helped demonstrate an edge case that I've updated `insertDynamicRequestHeaders` to manage). Let me know how it looks now.